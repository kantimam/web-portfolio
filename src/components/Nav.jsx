import React from 'react'
import GithubIcon from './GithubIcon'

const Nav = ({bgOpacity, scrollToPortfolio, scrollToContact, scrollTop}) => {
    if(!bgOpacity) return null
    return (
        <nav>
            <div className="inner flexCenterAll">
                <div onClick={scrollTop} className="mainIcon">KI</div>
                <ul>
                    <li onClick={scrollToPortfolio}>projects</li>
                    <li onClick={scrollToContact}>contact</li>
                </ul>
                <a className="githubLink" href="https://github.com/kantimam">
                    <GithubIcon/>
                </a>
            </div>
        </nav>
    )
}

export default Nav
