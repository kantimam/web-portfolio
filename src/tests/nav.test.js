import React from 'react';
import {
    render,
    cleanup,
    fireEvent
} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import Nav from '../components/Nav'

afterEach(cleanup);


test('does not render when invisible', () => {
    const { container } = render(<Nav ></Nav>)

    expect(container).toBeEmpty();
})


test('nav renders correctly', () => {

    const { getByText } = render(<Nav bgOpacity={0.6}></Nav>)

    expect(getByText("projects")).toBeInTheDocument()

    fireEvent.click(getByText("projects"));
})

test('click to work', () => {
    const portfolioFun = jest.fn();
    const contactFun = jest.fn();
    const { asFragment, getByText } = render(<Nav scrollToContact={contactFun} scrollToPortfolio={portfolioFun} bgOpacity={0.6}></Nav>)

    expect(getByText("projects")).toBeInTheDocument()

    expect(asFragment).toMatchSnapshot();

    fireEvent.click(getByText("projects"));

    expect(portfolioFun).toHaveBeenCalledTimes(1);
    expect(contactFun).not.toHaveBeenCalled();

})

