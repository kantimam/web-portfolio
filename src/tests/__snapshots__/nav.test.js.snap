// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`click to work 1`] = `[Function]`;
